.github/workflows/test.yml

```diff
@@ -0,0 +1,88 @@
+name: KBase Catalog test
+
+on:
+  pull_request:
+    types:
+    - opened
+    - reopened
+    - synchronize
+    - ready_for_review
+  push:
+    # run workflow when merging to main or develop
+    branches:
+      - main
+      - master
+      - develop
+
+jobs:
+  catalog_tests:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - python-version: '3.7'
+            mongo-version: '3.6'
+          - python-version: '3.7'
+            mongo-version: '7.0.4'
+    services:
+      mongo:
+        image: mongo:${{matrix.python-version}}
+        ports:
+            - 27017:27017
+        options: --name mongo${{matrix.python-version}}
+
+    steps:
+      - name: Repo checkout
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{matrix.python-version}}
+
+      - name: Install dependencies and set up test config
+        shell: bash
+        env:
+          KBASE_CI_TOKEN: ${{ secrets.KBASE_CI_TOKEN }}
+          ADMIN_USER: kbase_bot
```

Comment:
This shouldn't be hardcoded - check with one of the sysadmins to see if they can add an env var or secret for the token username

Reply:
👍

Reply:
:+1: